spec/models/spree/order_updater_spec.rb

```diff
@@ -86,4 +86,35 @@
       end
     end
   end
+
+  context 'when the set payment_state does not match the last payment_state' do
+    before { order.payment_state = 'previous_to_paid' }
+
+    context 'and the order is being updated' do
+      before { allow(order).to receive(:persisted?) { true } }
+
+      it 'creates a new state_change for the order' do
+        expect { updater.update_payment_state }
+          .to change { order.state_changes.size }.by(1)
+      end
+    end
+
+    context 'and the order is being created' do
+      before { allow(order).to receive(:persisted?) { false } }
+
+      it 'creates a new state_change for the order' do
+        expect { updater.update_payment_state }
```

Comment:
RSpec needs it to be a block. I guess that is because it can then be executed by RSpec later on and watch the changes on `order.state_changes.size`. See https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/change-matcher

Reply:
oh, I missed the change part... thanks!